samples/rag-storage-document-loaders/embed.py

```diff
@@ -0,0 +1,64 @@
+"""Sample showing embedding documents from Azure Blob Storage into Azure Search."""
+
+import os
+import warnings
+
+
+from azure.identity import DefaultAzureCredential
+from dotenv import load_dotenv
+from langchain_azure_ai.embeddings import AzureAIEmbeddingsModel
+from langchain_azure_ai.vectorstores import AzureSearch
+from langchain_community.document_loaders import PyPDFLoader
+
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+
+load_dotenv()
+warnings.filterwarnings("ignore", message=".*is currently in preview.*")
+warnings.filterwarnings("ignore", message=".*`AzureBlobStorageLoader` is in public preview.*")
+
+
+def main() -> None:
+    """Embed documents from Azure Blob Storage into Azure Search."""
+    loader = AzureBlobStorageLoader(
+        account_url=os.environ["AZURE_STORAGE_ACCOUNT_URL"],
+        container_name=os.environ["AZURE_STORAGE_CONTAINER_NAME"],
+        prefix=os.environ.get("AZURE_STORAGE_BLOB_PREFIX", None),
+        loader_factory=PyPDFLoader,  # Parses blobs as PDFs into LangChain Documents
+    )
+
+    embed_model = AzureAIEmbeddingsModel(
+        endpoint=os.environ["AZURE_EMBEDDING_ENDPOINT"],
+        credential=DefaultAzureCredential(),
```

Comment:
A little unfortunate that it won't allow us to show off the default azure credential but from a best practices perspective, it probably makes sense to create a single instance of the default credentials and provide it to each client instead of instantiate new ones. Mainly in reusing the credential object there is caching that we can leverage for overall perf improvements.